.github/workflows/build-tests.yml

```
@@ -54,3 +54,49 @@ jobs:
           *PR URL:* ${{ github.event.pull_request.url }}
       env:
         SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_SECRET }}
+
+  deploy:
+    name: deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Build Docker image
+      run: docker build --target production -t ${{ secrets.DOCKERHUB_USERNAME }}/todoapp:${{ github.sha }} -t ${{ secrets.DOCKERHUB_USERNAME }}/todoapp:latest .
+
+    - name: Push Docker image with SHA tag
+      run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todoapp:${{ github.sha }}
+
+    - name: Push Docker image with latest tag
+      run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todoapp:latest
```

Not that it particularly matters, but I believe we can combine these two statements if we want:
```suggestion
    - name: Push Docker image with both tags
      run: docker push ${{ secrets.DOCKERHUB_USERNAME }}/todoapp:${{ github.sha }} --all-tags
```